server/base/src/main/java/org/apache/accumulo/server/metadata/TabletMutatorBase.java

```diff
@@ -170,15 +170,15 @@ protected String getLocationFamily(LocationType type) {
   @Override
   public T putLocation(Location location) {
     Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
-    mutation.put(getLocationFamily(location.getType()), location.getSession(),
-        location.getHostPort());
+    mutation.put(getLocationFamily(location.getType()), "",
+        location.getServerInstance().toString());
```

Comment:
This may be one of the biggest changes in this PR. It removes the sessionId from the ColQ of the location columns in the metadata and puts the "<host:port>#<sessionId>#<resourceGroup>" in the Value. I was wondering why the sessionId was in the colq to begin with, that seems like it would allow more than one entry for the location type in the metadata table.

Reply:
Associated with this change, I ended up removing a test for https://issues.apache.org/jira/browse/ACCUMULO-1248 as it no longer applied - the test would not work.

Reply:
That's good information. I don't have the history to know why `sessionId` was in the column qualifier. It seemed like a bug to me as it would allow multiple `current`, `future`, or `last` entries.

Reply:
@EdColeman - do you happen to know where the code is that detects a change in sessionId?

Reply:
> I was wondering why the sessionId was in the colq to begin with, that seems like it would allow more than one entry for the location type in the metadata table.

I do not know of a specific reason why it was structured that way.  It did allow detection of certain bugs like if the manager set a location on a tablet that already had a location, but I don't think that is why it was structured that way.  Using conditional mutations can help avoid these bugs instead of detecting them after the fact.  We could take this a step further and push the current vs future designation into the value and have a single column for location.